### ISSUE TITLE
Zack's Tinkerer Journey

### ISSUE THREAD

**Issue by moul** [gnolang/hackerspace#2]

**Comment by schollz** [comment]
My name is Zack - my goal is to translate my love for Go and web2 into a love for Gno and web3. 

To reach this goal I am working on understanding the Gno language, what it represents, and how it is used. At the same time I am going to investigate the question "why web3?" and try to develop proof-of-concepts that make use of this new smart contract language. 

I'll post regularly here to update my progress. So for, in the Month of April:


1. I made a proof of concent and create an app for the Gno ecosystem: Microblog: https://github.com/schollz/gno/tree/microblog 
2. I prototyped an idea around generative audio with smart contracts: Bytebeat: https://github.com/schollz/gno/tree/bytebeat 
3. I opened a PR for a new realm: Microblog: https://github.com/gnolang/gno/pull/791 
4. I started to develop documentation to integrate into a future blogpost about how to get started with Gno: https://github.com/schollz/gno/blob/microblog/NOTES.md

In these starting steps I encountered small roadblocks, like a lack of `sort.Slice` (https://github.com/gnolang/gno/issues/750) and a lack of `encoding/binary` (https://github.com/gnolang/gno/issues/788). It seems both of these small issues are hinged on incorporating reflection into Gno which is an ongoing discussion. Living languages naturally have these roadblocks, and I can find ways to move around them while also engaging solutions. 

I am excited to prototype some new ideas, I'm starting to see some great potential for Gno.

**Comment by michelleellen** [comment]
Hey @schollz - 

Shall we get a date organized for the Microblog workshop and tutorial? That way we can add the tutorial to our upcoming docs, get the online workshop set up and promote it via the blog and socials.

cc: @zivkovicmilos

**Comment by schollz** [comment]
Yep! I'm happy to setup a date. I'm working on completing the changes for the PR for microblog right, process might take another week.

**Comment by michelleellen** [comment]
Update from the bi-weekly marketing sync for the month of June 
@schollz will be working on the tutorial for the "How to Build a Microblog" workshop and we'll record it the week of June 17th. We'll use opportunity at GopherCon and other channels to set up a live workshop session one week after GopherCon.  

cc: @moul

**Comment by michelleellen** [comment]
@schollz can you add here the docs for the workshop and tutorial 'Go to Gno: Generating Audio with Smart Contracts' for Thursday workshop?

**Comment by schollz** [comment]
yes! I will. I am working on them right now actually :). will upload them tonight or tomorrow.

**Comment by michelleellen** [comment]
Great, workshop is scheduled for Thursday the 12th at 11 am PST/8 pm CEST https://docs.google.com/forms/d/e/1FAIpQLSecrFQTJfofqofQ9qWx30EWffpa83A5yhwJbPtZlJ7lv__4aA/viewform

**Comment by schollz** [comment]
Great! Here is a link to the presentation materials: https://github.com/schollz/workshops-gno/tree/main/presentations/2023-10-09--generating-audio--schollz

Here is the current PDF: [gno-bytebeats-v1.pdf](https://github.com/gnolang/hackerspace/files/12859370/gno-bytebeats-v1.pdf)


I will make a PR to the main gnolang/workshops repo as soon as the slides are final. Please let me know if you need any more info!
